### ISSUE TITLE
Comparing object with "hasOwnProperty" keys fails

### ISSUE THREAD

**Issue by foolip** [mattphillips/deep-object-diff#58]
I'm trying to use deep-object-diff to compare different copies of https://github.com/mdn/browser-compat-data, which is a dataset describing the web platform itself. As such, it has [a key called "hasOwnProperty"](https://github.com/mdn/browser-compat-data/blob/d3e87462355e61d24aa9bf8398b0be2d1ef61306/javascript/builtins/Object.json#L930) to describe that method. deep-object-diff uses code like `obj.hasOwnProperty(key)` a lot, and `obj.hasOwnProperty` will in this case be an object in the BCD data, not `Object.prototype.hasOwnProperty`.

This causes deep-object-diff to throw an exception, the first place being here.
https://github.com/mattphillips/deep-object-diff/blob/45549c8225fc21c178dc6370e8028e4d08498e7b/src/diff/index.js#L12

Simplified a bit, here's a subset of the data:
```json
{
  "javascript": {
    "builtins": {
      "Object": {
        "hasOwnProperty": {
          "__compat": "an object with more stuff here"
        }
      }
    }
  }
}
```

Repro script showing the problem:
```js
const { diff } = require("deep-object-diff");
diff({"hasOwnProperty": 1}, {"hasOwnProperty": 2});
```

This will throw "Uncaught TypeError: r.hasOwnProperty is not a function".

A possible fix for this would be to add a `hasOwnProperty` wrapper to https://github.com/mattphillips/deep-object-diff/blob/master/src/utils/index.js and always use that, but there may be other ways.

**Comment by papb** [comment]
https://eslint.org/docs/rules/no-prototype-builtins would catch that.

**Comment by anko** [comment]
@foolip Review #59?

<sup>(Posting this because I think you don't get a notification when a PR links to your issue.)</sup>

**Comment by foolip** [comment]
@anko thanks for posting the fix, I've tried it and can confirm it fixes my problem.

**Comment by mattphillips** [comment]
Available in [v1.1.5](https://www.npmjs.com/package/deep-object-diff/v/1.1.5)
